channel.go

```diff
@@ -1290,6 +1290,12 @@ func (ch *Channel) ExchangeUnbind(destination, key, source string, noWait bool,
 	)
 }
 
+// GetNextPublishSeqNo returns the sequence number of the next message to be
+// published, when in confirm mode.
+func (ch *Channel) GetNextPublishSeqNo() uint64 {
+	return ch.confirms.published
```

Comment:
According to the description the function should return the sequence number (== deliveryTag?) of the next message that is published.
But if `GetNextPublishSeqNo` is called before the first `Publish()` call it returns 0, the deliveryTag of the first publishing is 1 though

Reply:
You are right @fho I am gonna make change as 
```
func (ch *Channel) GetNextPublishSeqNo() uint64 {
	ch.confirms.Lock()
	defer ch.confirms.Unlock()
	
	return ch.confirms.published + 1
}

Reply:
confirms.expected can't be used instead? I only had a quick look and was not sure :-)

Reply:
`published` incremented with every publish on the given channel. `expecting` increments, only when confirms are received as a part of resequencing and RabbitMQ broker does not guarantee that how soon a message will be confirmed. so `published` and `expecting` may be different at one moment and expecting may leg behind. Please correct me If I am wrong. :)

Reply:
Thank you for checking.  I agree, `expecting` is only incremented when a confirmation was received not for every publish call.